src/aspire/basis/basis.py

```diff
@@ -110,16 +113,13 @@ def evaluate_t(self, v):
                 f" Inconsistent dtypes v: {v.dtype} self: {self.dtype}"
             )
 
-        if not isinstance(v, Image) and not isinstance(v, Volume):
-            if self.ndim == 2:
-                _class = Image
-            elif self.ndim == 3:
-                _class = Volume
+        if not isinstance(v, self.cls):
             logger.warning(
```

Comment:
I'm considering whether we should start to use more python warnings. What do you think?